edb/schema/delta.py

```diff
@@ -1000,18 +1001,52 @@ def _build_alter_cmd_stack(
         root = DeltaRoot()
         return root, root
 
-    def _prohibit_if_expr_refs(
+    def _propagate_if_expr_refs(
         self,
         schema: s_schema.Schema,
         context: CommandContext,
         action: str,
-    ) -> None:
+    ) -> Tuple[s_schema.Schema, List[qlast.DDLCommand]]:
         scls = self.scls
         expr_refs = s_expr.get_expr_referrers(schema, scls)
+        # Commands to be executed after the original change is
+        # complete
+        finalize_ast: List[qlast.DDLCommand] = []
 
         if expr_refs:
             ref_desc = []
             for ref, fn in expr_refs.items():
+                from . import functions as s_func
+                from . import indexes as s_indexes
+                from . import pointers as s_pointers
+
+                if isinstance(ref, s_indexes.Index):
+                    # If the affected entity is an index, just drop it and
+                    # schedule it to be re-created.
+                    context.affected_finalization[self] = ref._create_command(
+                        schema, context)
+                    schema = ref.delete(schema)
+                    continue
+
+                elif isinstance(ref, (s_pointers.Pointer, s_func.Function)):
+                    # If the affected entity is a pointer, drop/create
+                    # the default value or computable. If the affected
+                    # entity is a function, alter it to drop/create the
+                    # body.
+                    delta_drop, cmd_drop = ref._get_command_stack(
+                        schema, context, AlterObject)
+                    delta_create, cmd_create = ref._get_command_stack(
+                        schema, context, AlterObject)
+
+                    # Copy own fields into the create command.
+                    value = ref.get_explicit_field_value(schema, fn, None)
+                    cmd_drop.set_attribute_value(fn, None)
```

Comment:
I don't think this will work in the general case. You cannot simply "drop" a body of a function object and same applies to the computables.  What you *can* do is replace the expression with something like `<exprtype>{}`.

Reply:
I've updated the code to use a dummy body for functions. It uses a string cast, selecting an object, or an `{}` with a cast (for collections, since we can't directly cast a `str` into an arbitrary array or tuple).